### ISSUE TITLE
absolute_path_linter only works on paths with single quotes

### ISSUE THREAD

**Issue by paulsromer** [jimhester/lintr#433]
When I run the absolute_path_linter on my file, it only catches paths that are wrapped in single quotes, not double quotes. 

Example file:
```{r}
test_path1 <- 'C:/Documents'
test_path2 <- "C:/Documents"
```

Output:
```{r}
lint("lintr_testfile.R", linters = with_defaults(absolute_path_linter()))
```
* `Line 1 Only use double-quotes.`
* `Line 1 Do not use absolute paths.`
With no flags for the absolute path on line 2. 

Looking at the source code for absolute_path_linter, I think the issue comes from this line which does the unquoting: `path <- unquote(token[["text"]], "'")`

**Comment by russHyde** [comment]
Also `absolute_path_linter` ignores absolute paths that are specified within `file.path()`:
```
ghi <- file.path("/", "dev", "null")
```

**Comment by russHyde** [comment]
Also, `nonportable_path_linter` appears to be affected by the single-quote vs double-quote issue

**Comment by russHyde** [comment]
I believe the bug that was originally identified (here and in #437 ) has been fixed by #442 . Would you (@paulsromer or @dominiqueemmanuel) be happy to review that pull request?
